homeassistant/components/bmw_connected_drive/button.py

```diff
@@ -1,72 +1,69 @@
 from bimmer_connected.vehicle import ConnectedDriveVehicle
 
 from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from . import (
-    DOMAIN as BMW_DOMAIN,
-    BMWConnectedDriveAccount,
-    BMWConnectedDriveBaseEntity,
-)
-from .const import CONF_ACCOUNT, DATA_ENTRIES
+from . import BMWConnectedDriveBaseEntity
+from .const import DOMAIN
+
+if TYPE_CHECKING:
+    from .coordinator import BMWDataUpdateCoordinator
 
 
 @dataclass
 class BMWButtonEntityDescription(ButtonEntityDescription):
     """Class describing BMW button entities."""
 
     enabled_when_read_only: bool = False
-    remote_function: Callable[
-        [ConnectedDriveVehicle], RemoteServiceStatus
-    ] | None = None
-    account_function: Callable[[BMWConnectedDriveAccount], None] | None = None
+    remote_function: str | None = None
+    account_function: Callable[[BMWDataUpdateCoordinator], Coroutine] | None = None
 
 
 BUTTON_TYPES: tuple[BMWButtonEntityDescription, ...] = (
     BMWButtonEntityDescription(
         key="light_flash",
         icon="mdi:car-light-alert",
         name="Flash Lights",
-        remote_function=lambda vehicle: vehicle.remote_services.trigger_remote_light_flash(),
+        remote_function="trigger_remote_light_flash",
     ),
     BMWButtonEntityDescription(
         key="sound_horn",
         icon="mdi:bullhorn",
         name="Sound Horn",
-        remote_function=lambda vehicle: vehicle.remote_services.trigger_remote_horn(),
+        remote_function="trigger_remote_horn",
     ),
     BMWButtonEntityDescription(
         key="activate_air_conditioning",
         icon="mdi:hvac",
         name="Activate Air Conditioning",
-        remote_function=lambda vehicle: vehicle.remote_services.trigger_remote_air_conditioning(),
+        remote_function="trigger_remote_air_conditioning",
     ),
     BMWButtonEntityDescription(
         key="deactivate_air_conditioning",
         icon="mdi:hvac-off",
         name="Deactivate Air Conditioning",
-        remote_function=lambda vehicle: vehicle.remote_services.trigger_remote_air_conditioning_stop(),
+        remote_function="trigger_remote_air_conditioning_stop",
     ),
     BMWButtonEntityDescription(
         key="find_vehicle",
         icon="mdi:crosshairs-question",
         name="Find Vehicle",
-        remote_function=lambda vehicle: vehicle.remote_services.trigger_remote_vehicle_finder(),
+        remote_function="trigger_remote_vehicle_finder",
     ),
     BMWButtonEntityDescription(
         key="refresh",
         icon="mdi:refresh",
         name="Refresh from cloud",
-        account_function=lambda account: account.update(),
+        account_function=lambda coordinator: coordinator.async_request_refresh(),
         enabled_when_read_only=True,
     ),
```

Comment:
This looks like it duplicates the functionality of `homeassistant.update_entity` https://www.home-assistant.io/docs/scripts/service-calls/#homeassistant-services

Reply:
Yes, it comes to down to the same thing (at least now that we're using a `DataUpdateCoordinator`).
Before, only the selected entity was updated using `update_entity`.
Now, the update forces all platforms to update (which is what the desired outcome of this refresh button was).

As it had been implemented since two releases and the original service will be removed in 2022.5 (#69808), should we add a deprecation warning and a breaking change section?

Reply:
Let's deprecate the button in the breaking changes section here and tell them to use `update_entity`.  Then remove the button in a future PR